code/addons/test/src/vitest-plugin/index.ts

```diff
@@ -82,56 +96,124 @@ export const storybookTest = (options?: UserOptions): Plugin => {
+        indexers,
+        docsOptions,
+        storybookEnv,
+        viteConfigFromStorybook,
+        previewLevelTagsData,
+      ] = await Promise.all([
+        //
+        presets.apply('framework', undefined),
+        presets.apply('stories'),
+        presets.apply('experimental_indexers', []),
+        presets.apply<DocsOptions>('docs', {}),
+        presets.apply('env', {}),
+        presets.apply('viteFinal', {}),
+
+        extractTagsFromPreview(configDir),
+      ]);
+
+      const generator = new StoryIndexGenerator(normalizeStories(storiesGlobsData, directories), {
         ...directories,
         indexers: indexers,
         docs: docsOptions,
-        workingDir,
       });
 
       await generator.initialize();
 
+      storiesGlobs = storiesGlobsData;
       storiesFiles = generator.storyFileNames();
+      previewLevelTags = previewLevelTagsData;
 
-      previewLevelTags = await extractTagsFromPreview(configDir);
-
-      const framework = await presets.apply('framework', undefined);
       const frameworkName = typeof framework === 'string' ? framework : framework.name;
-      const storybookEnv = await presets.apply('env', {});
-      const staticDirs = await presets.apply('staticDirs', []);
-
-      for (const staticDir of staticDirs) {
-        try {
-          statics.push(mapStaticDir(staticDir, configDir));
-        } catch (e) {
-          console.warn(e);
-        }
-      }
 
       // If we end up needing to know if we are running in browser mode later
       // const isRunningInBrowserMode = config.plugins.find((plugin: Plugin) =>
       //   plugin.name?.startsWith('vitest:browser')
       // )
 
-      config.test ??= {};
+      const baseConfig: Omit<ViteUserConfig, 'plugins'> = {
+        test: {
+          setupFiles: ['@storybook/experimental-addon-test/internal/setup-file'],
+
+          ...(finalOptions.storybookScript
+            ? {
+                globalSetup: ['@storybook/experimental-addon-test/internal/global-setup'],
+              }
+            : {}),
+
+          env: {
+            ...storybookEnv,
+            // To be accessed by the setup file
+            __STORYBOOK_URL__: storybookUrl,
+            __VITEST_INCLUDE_TAGS__: finalOptions.tags.include.join(','),
+            __VITEST_EXCLUDE_TAGS__: finalOptions.tags.exclude.join(','),
+            __VITEST_SKIP_TAGS__: finalOptions.tags.skip.join(','),
+          },
+
+          include: storiesFiles
+            .filter((path) => !path.endsWith('.mdx'))
+            .map((path) => convertPathToPattern(path)),
+
+          server: {
+            deps: {
+              inline: ['@storybook/experimental-addon-test'],
+            },
+          },
+
+          ...(input.test.browser
+            ? {
+                browser: {
+                  name: input.test.browser.name,
+                  screenshotFailures: input.test.browser?.screenshotFailures ?? false,
+                },
+              }
+            : {}),
```

Comment:
could this be simplified?

```suggestion
          // set screenshotFailures=false if not explicitly set
          ...(input.test?.browser && input.test.browser.screenshotFailures === undefined
            {
                browser: {
                  screenshotFailures: false,
                }),
```